fedimint-api-client/src/lib.rs

```diff
@@ -49,8 +48,8 @@ pub async fn try_download_client_config(invite_code: &InviteCode) -> anyhow::Res
 
     let query_strategy = FilterMap::new(
         move |cfg: ClientConfig| {
-            if federation_id.0 != cfg.global.api_endpoints.consensus_hash() {
-                bail!("Guardian api endpoint map does not hash to FederationId")
+            if federation_id != cfg.global.calculate_federation_id() {
```

Comment:
Why wouldn't want to try both here, so it works with old and new Federations all the same?